dataclasses_json/errors.py

```diff
@@ -0,0 +1,3 @@
+class DataclassSerializationException(KeyError):
```

Comment:
perhaps `class DataclassJsonError(Exception)` instead, to make it clear it's coming from this package, not dataclasses itself -- user-defined exceptions [typically inherit from `Exception`](https://docs.python.org/3/tutorial/errors.html#tut-userexceptions)

Reply:
It's weird not inheriting from `Exception` but I wanted to inherit from `KeyError` so that any client code that might have caught the old exception would still catch the new one.
I feel like inheriting from `Exception` would be a bit of a breaking change .. fine by me, but not sure if everyone would feel that way.

Reply:
gotcha, that's a good thought and thanks for clarifying